examples/subprojects/project_a/dobi.yaml

```diff
@@ -0,0 +1,22 @@
+#
+# project_a dobi file for subprojects example
+# See README.md for docs
+#
+
+#
+# Images
+#
+#
+
+image=project_a_img:
+    image: example/subprojects/project_a
+    context: project_a
```

Comment:
> But if you use the context of ., that is actually referring to the parent of project_a and project_b

That's right, but in this example you're including files from outside of the `project_x` directories, so that is necessary. You can use a context of `project_a/`, but then you won't have access to files outside of that context. Do you have a requirement to use files at the root of the repo, outside of the project directories?

> Ideally, I'd love for each sub-project's dobi.yaml file to be rooted in its own working directory by default.

That should be fine. We might need to fix the way includes are done, so that paths can be relative to the file which contains them, instead of the path of the root dobi.yaml.

> I got into this whole subproject thing because I was wanting to try to do cross-project coordination and building. 

I think that's completely reasonable, and I've heard this use case a few times from different teams, so I think it's relatively common for orgs doing SOA.